site-src/geps/gep-709.md

```diff
@@ -0,0 +1,260 @@
+
+The following example shows how a HTTPRoute in namespace foo could reference
+a Service in namespace bar. In this example a ReferencePolicy in the bar
+namespace explicitly allows references to Services from HTTPRoutes in the foo
+namespace.
+
+```yaml
+kind: HTTPRoute
+metadata:
+  name: foo
+  namespace: foo
+spec:
+  rules:
+  - matches:
+    - path: /bar
+    forwardTo:
+      backend:
+      - name: bar
+        namespace: bar
+---
+kind: ReferencePolicy
+metadata:
+  name: bar
+  namespace: bar
+spec:
+  from:
+  - group: networking.gateway.k8s.io
+    kind: HTTPRoute
+    namespace: foo
+  to:
+  - group: core
+    kind: Service
+```
+
+### API
+This proposed API is fairly straightforward, but comes with a couple notable
+decisions:
+
+1. Each ReferencePolicy only supports a single From and To section. Additional
+   trust relationships can be modeled with additional ReferencePolicy resources.
+1. Resource names are intentionally excluded from this policy for simplicity and
+   because they rarely provide any meaningful protection.
+1. A single Namespace is allowed per "From" struct. Although a selector would be
+   more powerful it may encourage unnecessarily insecure configuration.
+
+```go
+// ReferencePolicy identifies cross namespace relationships that are trusted for
+// Gateway API.
+type ReferencePolicy struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+
+    // Spec defines the desired state of ReferencePolicy.
+    Spec ReferencePolicySpec `json:"spec,omitempty"`
+}
+
+
+// ReferencePolicySpec identifies a cross namespace relationship that is trusted
+// for Gateway API.
+type ReferencePolicySpec struct {
+    // From describes the trusted namespaces and kinds that can reference the
+    // resources described in "To".
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinItems=1
+    From []ReferencePolicyFrom `json:"from"`
+
+    // To describes the resources that may be referenced by the resources
+    // described in "From".
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinItems=1
+    To []ReferencePolicyTo `json:"to"`
+}
+
+// ReferencePolicyFrom describes trusted namespaces and kinds.
+type ReferencePolicyFrom struct {
+    // Group is the group of the referrent.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    Group string `json:"group"`
+
+    // Kind is kind of the referrent.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    Kind string `json:"kind"`
+
+    // Namespace is the namespace of the referrent.
+    //
+    // Support: Core
+    //
+    // +kubebuilder:validation:MinLength=1
+    // +kubebuilder:validation:MaxLength=253
+    Namespace string `json:"namespace,omitempty"`
+}
+
+// ReferencePolicyTo describes trusted kinds.
```

Comment:
"ReferencePolicyTo describes what Kinds are allowed as targets of the references."

Reply:
(Not a trusted kind, it's object that we own anyway)

Reply:
Much better, thanks!